### DISCUSSION TITLE
Using QTH within custom component.

### DISCUSSION THREAD

**Discussion by mtn090899** [quasarframework/quasar#12960] [General - Components / Directives / etc]
I am utilizing the q-table component to make a custom component. I'm trying to utilize the q-th component within said custom component, but there seems to be an issue with reading the sortable property.  The data I am using is the example data from the q-table documentation.  Which is a bit odd, since if I do similar code just utilizing q-table and not wrapping it within a custom component, it works just fine. This is the error message. 

TypeError: Cannot read properties of undefined (reading 'sortable')
    at Proxy.eval (QTh.js?357e:40:1)

I honestly don't know where to tackle the problem, if there is an answer to this question, please let me know. For further elaboration as to what Im trying to do, I'm trying to place an icon next to the headers, perhaps a clickable icon. Things of that nature. Thank you.

Use of custom component:

``` 
<template>
<q-page>
<h-table title="Title"  :columns="columns"  :rows ="rows" row-key="name"  class="q-pa-xl"></h-table>
</q-page>
</template>

<script>
{
        return {
        columns,
        rows,
        }

    },
</script>
``` 

Custom component: 

```
<template>
      <q-table
      :title="title"
      :rows="rows"
      :columns="columns"
      :row-key="row_key"
      :table-header-style="{ backgroundColor: '#DCEEFB' }"
    >


     <template v-slot:header="props">
                 <q-th :props="props">

        </q-th>
      </template>
    </q-table>
</template>

<script>
export default {
    props: {
        columns: {
            type: Array
        },
        rows: {
            type: Array
        },
        title: {
            type: String
        },
        row_key: {
            type: String
        }
    
}
}
</script>

```

Columns and Rows Data:

const columns = [
  {
    name: 'name',
    required: true,
    label: 'Dessert (100g serving)',
    align: 'left',
    field: row => row.name,
    format: val => `${val}`,
    sortable: true
  },
  { name: 'calories', align: 'center', label: 'Calories', field: 'calories', sortable: true },
  { name: 'fat', label: 'Fat (g)', field: 'fat', sortable: true },
  { name: 'carbs', label: 'Carbs (g)', field: 'carbs' },
  { name: 'protein', label: 'Protein (g)', field: 'protein' },
  { name: 'sodium', label: 'Sodium (mg)', field: 'sodium' },
  { name: 'calcium', label: 'Calcium (%)', field: 'calcium', sortable: true, sort: (a, b) => parseInt(a, 10) - parseInt(b, 10) },
  { name: 'iron', label: 'Iron (%)', field: 'iron', sortable: true, sort: (a, b) => parseInt(a, 10) - parseInt(b, 10) }
]

const rows = [
  {
    name: 'Frozen Yogurt',
    calories: 159,
    fat: 6.0,
    carbs: 24,
    protein: 4.0,
    sodium: 87,
    calcium: '14%',
    iron: '1%'
  },
  {
    name: 'Ice cream sandwich',
    calories: 237,
    fat: 9.0,
    carbs: 37,
    protein: 4.3,
    sodium: 129,
    calcium: '8%',
    iron: '1%'
  },
  {
    name: 'Eclair',
    calories: 262,
    fat: 16.0,
    carbs: 23,
    protein: 6.0,
    sodium: 337,
    calcium: '6%',
    iron: '7%'
  },
  {
    name: 'Cupcake',
    calories: 305,
    fat: 3.7,
    carbs: 67,
    protein: 4.3,
    sodium: 413,
    calcium: '3%',
    iron: '8%'
  },
  {
    name: 'Gingerbread',
    calories: 356,
    fat: 16.0,
    carbs: 49,
    protein: 3.9,
    sodium: 327,
    calcium: '7%',
    iron: '16%'
  },
  {
    name: 'Jelly bean',
    calories: 375,
    fat: 0.0,
    carbs: 94,
    protein: 0.0,
    sodium: 50,
    calcium: '0%',
    iron: '0%'
  },
  {
    name: 'Lollipop',
    calories: 392,
    fat: 0.2,
    carbs: 98,
    protein: 0,
    sodium: 38,
    calcium: '0%',
    iron: '2%'
  },
  {
    name: 'Honeycomb',
    calories: 408,
    fat: 3.2,
    carbs: 87,
    protein: 6.5,
    sodium: 562,
    calcium: '0%',
    iron: '45%'
  },
  {
    name: 'Donut',
    calories: 452,
    fat: 25.0,
    carbs: 51,
    protein: 4.9,
    sodium: 326,
    calcium: '2%',
    iron: '22%'
  },
  {
    name: 'KitKat',
    calories: 518,
    fat: 26.0,
    carbs: 65,
    protein: 7,
    sodium: 54,
    calcium: '12%',
    iron: '6%'
  }
]

**Comment by metalsadman** [comment]
`row-key="name"` should be `row_key`, like how you registered it in your props.

**Reply by mtn090899** [reply]
Hi, 

Thank you for answering, however that didn't really change the results with our without the utilization of the q-th component. I'm still receiving the same error, is there anything else wrong that you could see that would possibly be causing this?

Thank you!

**Reply by metalsadman** [reply]
You can put up a codepen, so peeps can play with it.
